index.js

```diff
@@ -1,15 +1,59 @@
 var $ = require('jquery')
-var Mustache = require('mustache')
+var Handlebars = require('handlebars')
 
 function AnnotationPoller (opts) {
+  this._installExtensions()
+
   this.pollInterval = opts.pollInterval || 3000
   this.pkg = opts.pkg // what package should we load annotations for?
   this.endpoint = '/api/v1/annotations/' + this.pkg
   this.annotations = {}
-  this.template = '<li id="annotation-{{id}}" style="{{status}}" data-fingerprint={{fingerprint}}><span>{{description}}</span><a href="{{{external-link}}}">{{external-link-text}}</a></li>'
+  this.template = Handlebars.compile(
+    '<li id="annotation-{{id}}" style="{{status}}" data-fingerprint={{fingerprint}}>' +
+    '<ul class="addon-container">' +
+    '  <li><h3>{{title}}</h3></li>' +
+    '  {{#each rows}}' +
+    '    <li>' +
+    '    {{#hasKey this "image"}}' +
+    '      <img src="{{{image.url}}}" alt="{{image.text}}" />' +
+    '    {{/hasKey}}' +
+    '    {{#hasKey this "link"}}' +
+    '      {{#isArray this "link"}}' +
+    '        {{#each link}}' +
+    '          <a href="{{{url}}}">{{text}}</a>{{#unless @last}},{{/unless}}' +
+    '        {{/each}}' +
+    '        {{else}}' +
+    '          <a href="{{{link.url}}}">{{link.text}}</a>' +
+    '     {{/isArray}}' +
+    '     {{/hasKey}}' +
+    '     {{#hasKey this "text"}}' +
+    '       <span>{{{text}}}</span>' +
+    '     {{/hasKey}}' +
+    '  </li>' +
+    '  {{/each}}' +
+    '</ul>' +
+  '</li>')
```

Comment:
Could possibly use a _template literal_ to define a multi-line string here instead of concatenating (assuming Node 4+):

```js
this.template = Handlebars.compile(`
  <li id="annotation-{{id}}" style="{{status}}" data-fingerprint={{fingerprint}}>
    <ul class="addon-container">
    ...
    </ul>
  </li>
`)
```

Won't be exactly the same string since line breaks will be preserved, but might make for easier reading/maintenance.